extensions/blocks/premium-content/_inc/access-check.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Determine access to premium content.
+ *
+ * @package Automattic\Jetpack\Extensions\Premium_Content
+ */
+
+namespace Automattic\Jetpack\Extensions\Premium_Content;
+
+require __DIR__ . '/subscription-service/include.php';
+
+/**
+ * Determines if the memberships module is set up.
+ *
+ * @return bool Whether the memberships module is set up.
+ */
+function pre_render_checks() {
+	// If Jetpack is not yet configured, don't show anything ...
+	if ( ! class_exists( '\Jetpack_Memberships' ) ) {
+		return false;
+	}
+	// if stripe not connected don't show anything...
+	if ( empty( \Jetpack_Memberships::get_connected_account_id() ) ) {
+		return false;
+	}
+	return true;
+}
```

Comment:
There is an edge case here where the site (Jetpack or wpcom) does not have a sufficient plan ugprade, but _does_ have a Stripe account connected. The user will get the upgrade nudge in the edtior, but since we only check for the Stripe account in the `pre_render_checks`, the block will render just fine on the frontend. I believe this is a pre-existing problem.